Doist/reactist
# feat: TextArea autoexpand

## Short description

Adds support for the `TextArea` to auto-expand its height to fit the content as you type. Based on [this approach](https://css-tricks.com/the-cleanest-trick-for-autogrowing-textareas/), without requiring any external library or complex JS. It seems to work pretty well.

## References

- https://twist.com/a/1585/ch/140545/t/3919781/c/80968838
- https://twist.com/a/1585/ch/1569/t/4025021/c/80946449

## Test plan

- Open the TextArea stories
    - [x] check that in the first 3 stories, the text area behaves and looks as before (compare with https://doist.github.io/reactist/ if needed)
    - [x] Try changing some props such as the variant, tone, etc. The text area still reacts to the prop changes as expected.
- Open the new "Auto Grow" story, and check that the two text areas there behave each as expected.
    - [x] The top one auto-expands as you type or paste text. On the other hand, it does not allow you to resize it manually.
    - [x] The one at the bottom does not auto-expand. However, it allows you to resize it vertically as you see fit.

## PR Checklist



-   [x] Added tests for bugs / new features
-   [x] Updated docs (storybooks, readme)
-   [x] Executed `npm run validate` and made sure no errors / warnings were shown
-   [x] Described changes in `CHANGELOG.md`
-   [x] Bumped version in `package.json` and `package-lock.json` (`npm --no-git-tag-version version <major|minor|patch>`) [ref](https://docs.npmjs.com/cli/v6/commands/npm-version)
-   [x] Updated all static build artifacts (`npm run build-all`)

## Versioning

New minor release.
